### PR TITLE
feat: Cleanup old data on import

### PR DESCRIPTION
**Pull request by marcometz**
- remove old data on daily imports
- only import future calendar dates

MQGB-48

### CODE REVIEW

File: app/services/gtfs_redis_adapter.rb [hunk 1]
```diff
@@ -57,6 +70,13 @@ def set_gtfs_stop_times(stop_id, stop_times, data_provider_id)
     redis.rpush("#{namespace}:gtfs:#{data_provider_id}:stop:#{stop_id}:stop_times", stop_times.to_json)
   end
 
+  # delete all key starting with stop
+  def delete_all_stop_times(data_provider_id)
+    redis.scan_each(match: "#{namespace}:gtfs:#{data_provider_id}:stop:*") do |key|
```
Comment (donni106) [inline app/services/gtfs_redis_adapter.rb:75]: `stop_times` instead of `stop`?

File: app/services/public_transportation/import_service.rb [hunk 2]
```diff
@@ -26,6 +26,7 @@ def initialize(gtfs_url, data_provider_id, whitelist_poi = [])
     @whitelist = whitelist_poi.map(&:to_i) if whitelist_poi.present?
     @pois = {}
     @redis = GtfsRedisAdapter.new
+    @import_from_calender_date = (Date.today - (1.day)).strftime("%Y%m%d")
```
Comment (donni106) [inline app/services/public_transportation/import_service.rb:29]: We do not need `- 1.day` or am I wrong? today at 00:00:00 would be fine I guess.

File: app/services/public_transportation/import_service.rb [hunk 3]
```diff
@@ -93,7 +94,13 @@ def call
     def parse_calendar_dates(zip_file)
       p "Parsing calendar_dates.txt"
       calendar_dates_counter = 0
+
+      # cleanup stop_times bevor import
+      @redis.delete_all_gtfs_calendar_dates(@data_provider_id)
```
Comment (donni106) [inline app/services/public_transportation/import_service.rb:99]: cleanup stop times and than calling deletion of calendar dates? what about `# cleanup before import` as comment without specifying?

File: app/services/public_transportation/import_service.rb [hunk 4]
```diff
@@ -108,6 +115,10 @@ def parse_calendar_dates(zip_file)
     def parse_calendar(zip_file)
       p "Parsing calendar.txt"
       calendar_counter = 0
+
+      # cleanup stop_times bevor import
+      @redis.delete_all_gtfs_calendar(@data_provider_id)
```
Comment (donni106) [inline app/services/public_transportation/import_service.rb:120]: cleanup stop times and then calling deletion of calendar dates? what about `# cleanup before import` as comment without specifying?

File: app/services/gtfs_redis_adapter.rb (hunk 1 above)
Comment (marcometz) [reply app/services/gtfs_redis_adapter.rb:75]: the redis key is "#{namespace}:gtfs:#{data_provider_id}:stop:#{stop_id}:stop_times"

so we by deleting "#{namespace}:gtfs:#{data_provider_id}:stop:*"
we delete all "#{stop_id}:stop_times" => all stop_times for all stops

the stops itself are stored in SQL DB and not in redis

File: app/services/public_transportation/import_service.rb (hunk 3 above)
Comment (marcometz) [reply app/services/public_transportation/import_service.rb]: updated comments

File: app/services/gtfs_redis_adapter.rb (hunk 1 above)
Comment (donni106) [reply app/services/gtfs_redis_adapter.rb:75]: ok

File: app/services/public_transportation/import_service.rb (hunk 2 above)
Comment (donni106) [reply app/services/public_transportation/import_service.rb:29]: ok

File: app/services/public_transportation/import_service.rb (hunk 3 above)
Comment (donni106) [reply app/services/public_transportation/import_service.rb]: would you like to change the `bevor`s to `before`s? ;)

Comment (marcometz) [reply app/services/public_transportation/import_service.rb]: thanks!

**Review by donni106 — APPROVED** [review]

**Pull request merged**